node/node_test.go

```diff
@@ -622,6 +623,50 @@ func TestManualConditionsPreserved(t *testing.T) {
 
 	t.Log(newNode.Status.Conditions)
 }
+
+func TestNodePingSingleInflight(t *testing.T) {
+	const pingTimeout = 100 * time.Millisecond
+	c := testclient.NewSimpleClientset()
+	testP := &testNodeProviderPing{}
+
+	var calls, finished int64
+	testP.customPingFunction = func(context.Context) error {
+		atomic.AddInt64(&calls, 1)
+		time.Sleep(2 * time.Second)
```

Comment:
Could we use some synchronization here instead of sleeps?

It seems like you want this to timeout.
So it could wait for `<-ctx.Done()`.
And/or it could wait for a singal after the first result returns.

Reply:
I'll clean this up.